amp/WEB-INF/src/org/digijava/kernel/ampapi/endpoints/activity/preview/PreviewActivityService.java

```diff
@@ -292,5 +308,108 @@ private Double convertProjectCostAmount(AmpFundingAmount projectCost, String cur
+            String ownerInfo = TranslatorWorker.translateText("Workspace where the activity was created");
+            previewWorkspaces.add(new PreviewWorkspace(ownerTeam.getName(), PreviewWorkspace.Type.TEAM, ownerInfo));
+    
+            AmpTeam parentTeam = ownerTeam.getParentTeamId();
+            String path = ownerTeam.getName();
+            
+            while (parentTeam != null) {
+                path = String.format("%s -> %s", path, parentTeam.getName());
+                previewWorkspaces.add(new PreviewWorkspace(parentTeam.getName(),
+                        PreviewWorkspace.Type.MANAGEMENT, path));
+                
+                parentTeam = parentTeam.getParentTeamId();
+            }
+    
+            List<AmpTeam> computedTeams = TeamUtil.getAllTeams().stream()
+                    .filter(t -> Boolean.TRUE.equals(t.getComputation()) || Boolean.TRUE.equals(t.getUseFilter()))
+                    .filter(t -> t.getAmpTeamId() != ownerTeam.getAmpTeamId())
+                    .collect(Collectors.toList());
+            
+            final StringBuffer wsQueries = new StringBuffer();
+            for (AmpTeam team : computedTeams) {
+                String wsQuery = WorkspaceFilter.generateWorkspaceFilterQueryForTeam(team.getAmpTeamId());
+
+                if (wsQueries.length() > 0) {
+                    wsQueries.append(" UNION ");
+                }
+
+                wsQueries.append(AmpMessageWorker.addTeamIdToQuery(wsQuery, team.getAmpTeamId(), team.getName()));
+            }
 
+            final Map<Long, List<Team>> activityTeams = new HashMap<>();
+
+            PersistenceManager.getSession().doWork(conn -> {
+                RsInfo teamsInActivityQuery = SQLUtils.rawRunQuery(conn, wsQueries.toString(), null);
+                while (teamsInActivityQuery.rs.next()) {
+                    // activityTeams
+                    Long ampActivityId = teamsInActivityQuery.rs.getLong(1);
+                    if (activityTeams.get(ampActivityId) == null) {
+                        activityTeams.put(ampActivityId, new ArrayList<>());
+                    }
+                    activityTeams.get(ampActivityId).add(
+                            new Team(teamsInActivityQuery.rs.getLong(2),
+                                    teamsInActivityQuery.rs.getString(TEAM_NAME_INDEX)));
+                }
+                teamsInActivityQuery.close();
+            });
+            
+            if (activityTeams.containsKey(activityId)) {
+                for (Team team : activityTeams.get(activityId)) {
+                    previewWorkspaces.add(new PreviewWorkspace(team.getTeamName(), PreviewWorkspace.Type.COMPUTED));
+                }
+            }
+            
+        } catch (DgException e) {
+            throw new RuntimeException(e);
+        }
+        
+        return previewWorkspaces;
+    }
+    
+    public boolean isViewWorskpacesButtonVisible(Long ampActivityId) {
```

Comment:
we have similar functionality here
ActivityUtil,canValidateAcitivty (please fix the typo) 
I think we can refactor since they have common functionality (if its manager / approver)